R/utils.R

```diff
@@ -150,3 +144,64 @@ update_field_names <- function(db_metadata, raw_or_label = 'raw'){
     unnest(cols = field_name_updated)
 }
 
+#' Update Multiple Choice Fields with Label Data
+#'
+#' @param db_data A REDCap database object
+#' @param db_metadata A REDCap metadata object
+#'
+#' @import dplyr
+#' @keywords internal
+
+multi_choice_to_labels <- function(db_data, db_metadata){
+
+  for (i in 1:nrow(db_metadata)) {
+
+    # Extract metadata field name and database corresponding column name
+    field_name <- db_metadata$field_name_updated[i]
+
+    # yesno datatype handling ----
+    if (db_metadata$field_type[i] == "yesno") {
+      db_data <- db_data %>%
+        mutate(
+          across(.cols = field_name, .fns = ~case_when(. == 1 ~ TRUE, . == 0 ~ FALSE, TRUE ~ NA))
+        )
+    }
+
+    # truefalse datatype handling ----
+    if (db_metadata$field_type[i] == "truefalse") {
+      db_data <- db_data %>%
+        mutate(
+          across(.cols = field_name, .fns = ~case_when(. == 1 ~ TRUE, . == 0 ~ FALSE, TRUE ~ NA))
+        )
+    }
+
+    # checkbox datatype handling ----
+    if (db_metadata$field_type[i] == "checkbox") {
+      db_data <- db_data %>%
+        mutate(
+          across(.cols = field_name, .fns = ~case_when(. == 1 ~ TRUE, . == 0 ~ FALSE, TRUE ~ NA))
+        )
+    }
+
+    # dropdown and radio datatype handling ----
+    if (db_metadata$field_type[i] == "dropdown" | db_metadata$field_type[i] == "radio") {
+
+      # Retrieve parse_labels key for given field_name
+      parse_labels_output <- parse_labels(db_metadata$select_choices_or_calculations[i])
+
+      # Replace values from db_data$(field_name) with label values from  parse_labels key
+      db_data <- db_data %>%
+        left_join(parse_labels_output %>% rename(!!field_name := raw), # Could not get working in by argument, instead inject field name for rename in parse_labels_output
```

Comment:
I don't think the `rename` is necessary if you use a named `by` -- i.e. `by = c("raw" = field_name)` or similar